src/app/BasicRule.h

```diff
@@ -0,0 +1,49 @@
+#pragma once
+
+#include <regex>
+
+#include "RuleManager.h"
+
+namespace SAIL { namespace rule {
+
+class BasicRule : public RuleModule
+{
+private:
+    struct ptrace_arg_cmp {
+        unsigned int paraIndex;
+        std::string action;
+        int intValue;
+        std::vector<int> bytesValue;
+        std::string strValue;
+
+        ptrace_arg_cmp(const unsigned int paraIndex, const std::string &action, const int intValue) : 
+            paraIndex(paraIndex), action(action), intValue(intValue) {}
+        ptrace_arg_cmp(const unsigned int paraIndex, const std::string &action, const std::vector<int> &bytesValue) : 
+            paraIndex(paraIndex), action(action), bytesValue(bytesValue) {}
+        ptrace_arg_cmp(const unsigned int paraIndex, const std::string &action, const std::string &strValue) : 
+            paraIndex(paraIndex), action(action), strValue(strValue) {}
+    };
+
+    struct Rule {
+        int sysnum;
+        int id;
+        std::vector<struct scmp_arg_cmp> specs;
+        std::vector<struct ptrace_arg_cmp> pSpecs;
+        bool needExtraCheck; // if action is matchBytes or matchRe, which seccomp cannot handle, need extra check with ptrace
```

Comment:
extraCheck is useful but not good enough.
the range of "extra" is not presented.

You should define the supported action explicitly.